navigation/navigation-runtime/src/webMain/kotlin/androidx/navigation/BrowserHistory.kt

```diff
@@ -178,10 +178,10 @@ private fun NavBackStackEntry.getRouteWithArgs(): String? {
 }
 
 private fun NavBackStackEntry.getRouteAsUrlFragment() =
-    getRouteWithArgs()?.let { r -> "#$r" }.orEmpty()
+    getRouteWithArgs()?.let { r -> "#${encodeURIComponent(r)}" }.orEmpty()
 
 private fun NavController.tryToNavigateToUrlFragment(localWindow: BrowserWindow) {
-    val route = localWindow.location.hash.substringAfter('#', "")
+    val route = decodeURIComponent(localWindow.location.hash.substringAfter('#', ""))
```

Comment:
in theory if in all browsers hash is returned with # we can just remove first symbol but not a big deal

Reply:
why? I don't see benefits here. It is not faster